sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -165,13 +168,13 @@ export class IER12ProcessingService {
 
   /**
    * Create the Request content for the IER 12 file by populating the content.
-   * @param pendingAssessment pending assessment of institutions.
+   * @param application application and its current assessment.
    * @returns IER 12 records for the student assessment.
    */
   private async createIER12Record(
-    pendingAssessment: StudentAssessment,
+    application: Application,
   ): Promise<IER12Record[]> {
-    const application = pendingAssessment.application;
+    const pendingAssessment = application.currentAssessment;
```

Comment:
I believe it is not necessary to update this method to accept application, it can still receive assessment from the consumer.

Reply:
It is slightly confusing with IER. Even though we get the assessment wise data, it is prepared disbursement wise.  

Please look at the reference here. https://github.com/bcgov/SIMS/blob/4a8e27624fe7eea6b97772bf394ff07832f5feb5/sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts#L218 

if there is no disbursement schedule, then empty array is returned by `createIER12Record` method.

Reply:
I am not sure if I followed, but the behavior is still the same, just the application data comes from a different object now.
It no longer comes from inside the assessment; hence, the variables were adjusted, but disbursement behavior is not supposed to be affected.